libs/ui/src/components/atoms/input/input.tsx

```diff
@@ -1,3 +1,31 @@
-const Input = () => <input />;
+import { InputType } from './types';
+import classNames from 'classnames';
+import { BaseInput, WithIcon } from './style';
+
+export const Input: InputType = (props) => {
+  const {
+    long,
+    full,
+    icon,
+    warning,
+    error,
+    placeholder = 'Placeholder',
+    number,
+  } = props;
+
+  const classes = classNames({ long, full, warning, error, number });
+
+  if (icon) {
+    const Icon = icon;
+    return (
+      <WithIcon className={classes}>
+        <BaseInput {...props} className={classes} placeholder={placeholder} />
+        <Icon size="2rem" />
+      </WithIcon>
```

Comment:
Does this mean that the icon and base input styling would be the same? Is a potential for those to diverge?

Reply:
Becuase the input's width is dependant on the parent's width, they actually need to work in unison. Don't need to diverge for now, but if we need to we can always generate separate class strings.

Reply:
Ah, I see why it might suck. I'll just do this now.